src/main/java/seedu/duke/commands/TransactionCommand.java

```diff
@@ -0,0 +1,161 @@
+     * Optional flags allow the user to search by date, month, description or currency
+     *
+     * @param ui       The instance of the UI class
+     * @param accounts The instance of the AccountList class
+     */
+    @Override
+    public void execute (Ui ui, AccountList accounts) {
+        String[] args = input.split(" ");
+        String transactionsString;
+        try {
+            if (args.length == 1) {
+                // Print all transactions
+                printAllTransactions(ui);
+                return;
+            }
+            // At this point there is an additional flag in user input signalling a search parameter
+            String flag = args[1];
+            printTransactionsByFlag(ui, args, flag);
+        } catch (NoTransactionsRecordedException e) {
+            ui.printMessage(ErrorMessage.NO_TRANSACTIONS_RECORDED);
+        } catch (InvalidSearchTransactionByDescException e) {
+            ui.printMessage(ErrorMessage.INVALID_TRANSACTIONS_OF_DESC);
+        } catch (InvalidSearchTransactionByCurrencyException e) {
+            ui.printMessage(ErrorMessage.INVALID_TRANSACTIONS_OF_CURRENCY);
+        } catch (InvalidSearchTransactionByDateException e) {
+            ui.printMessage(ErrorMessage.INVALID_TRANSACTIONS_OF_DATE);
+        } catch (InvalidSearchTransactionByMonthException e) {
+            ui.printMessage(ErrorMessage.INVALID_TRANSACTIONS_OF_MONTH);
+        } catch (NoTransactionsOfSearchParameterException e) {
+            ui.printMessage(ErrorMessage.NO_TRANSACTIONS_FOUND);
+        } catch (InvalidTransactionFlagException e) {
+            ui.printMessage(ErrorMessage.INVALID_TRANSACTION_FLAG);
+        } catch (IllegalArgumentException e) {
+            ui.printMessage(ErrorMessage.INVALID_CURRENCY);
+        }
+    }
+
+    /**
+     * Method that prints all transactions in the transaction list
+     *
+     * @param ui The instance of the UI class
+     * @throws NoTransactionsRecordedException if there are no transactions in the list
+     */
+    private static void printAllTransactions (Ui ui) throws NoTransactionsRecordedException {
+        TransactionManager transactionManager = TransactionManager.getInstance();
+        String transactionsString;
+        transactionsString = transactionManager.getAllTransactionsString();
+        ui.printMessage(Message.SHOW_ALL_TRANSACTIONS_HEADER.getMessage());
+        ui.printMessage(transactionsString);
+    }
+
+    /**
+     * Method that searches a string for a substring
+     * Upon finding that substring, returns the rest of the string after that substring,
+     * excluding the searched substring
+     *
+     * @param stringToSearch The string to search within
+     * @param substring      The substring to search for
+     * @return The rest of the input string after and excluding the input substring, null if substring is not found
+     */
+    private String findSubstringToEndAfterASubstring (String stringToSearch, String substring) {
```

Comment:
Perhaps we could rename this function to `substringAfterKeyword` so its purpose is a bit clearer?

Reply:
Method renamed to substringAfterKeyword